models/user_conversation_mapping.go

```diff
@@ -36,13 +42,63 @@ const (
 	mappingsTable string = "users_to_conversations"
 )
 
+// Merge creates a new UserConversationMapping by copying the original mapping
+// and replacing its fields with the non-zero-value fields of a patch mapping
+func (m *UserConversationMapping) Merge(patch *UserConversationMapping) *UserConversationMapping {
```

Comment:
Shouldn't `Pending` and `LastOpened` be updated too?

Reply:
Yeah they should be.